pkg/subctl/cmd/utils/restconfig/restconfig.go

```diff
@@ -0,0 +1,162 @@
+
+	"github.com/submariner-io/admiral/pkg/resource"
+	subv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+
+	"github.com/submariner-io/submariner-operator/apis/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils"
+)
+
+type RestConfig struct {
+	Config      *rest.Config
+	ClusterName string
+}
+
+func GetMultipleRestConfigs(kubeConfigPath string, kubeContexts []string) ([]RestConfig, error) {
+	var restConfigs []RestConfig
+
+	rules := clientcmd.NewDefaultClientConfigLoadingRules()
+	rules.DefaultClientConfig = &clientcmd.DefaultClientConfig
+	overrides := &clientcmd.ConfigOverrides{ClusterDefaults: clientcmd.ClusterDefaults}
+	rules.ExplicitPath = kubeConfigPath
+
+	contexts := []string{}
+	if len(kubeContexts) > 0 {
+		contexts = append(contexts, kubeContexts...)
+	} else {
+		kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides)
+		rawConfig, err := kubeConfig.RawConfig()
+		if err != nil {
+			return restConfigs, err
+		}
+		for context := range rawConfig.Contexts {
+			contexts = append(contexts, context)
+		}
+	}
+
+	for _, context := range contexts {
+		if context != "" {
+			overrides.CurrentContext = context
+			config, err := GetClientConfigAndClusterName(rules, overrides)
+			if err != nil {
+				return nil, err
+			}
+
+			restConfigs = append(restConfigs, config)
+		}
+	}
+
+	return restConfigs, nil
+}
+
+func GetBrokerRestConfigAndNamespace(submariner *v1alpha1.Submariner,
+	serviceDisc *v1alpha1.ServiceDiscovery) (*rest.Config, string, error) {
+	if submariner != nil {
+		// Try to authorize against the submariner Cluster resource as we know the CRD should exist and the credentials
+		// should allow read access.
+		restConfig, _, err := resource.GetAuthorizedRestConfig(submariner.Spec.BrokerK8sApiServer, submariner.Spec.BrokerK8sApiServerToken,
+			submariner.Spec.BrokerK8sCA, rest.TLSClientConfig{}, schema.GroupVersionResource{
+				Group:    subv1.SchemeGroupVersion.Group,
+				Version:  subv1.SchemeGroupVersion.Version,
+				Resource: "clusters",
+			}, submariner.Spec.BrokerK8sRemoteNamespace)
+
+		return restConfig, submariner.Spec.BrokerK8sRemoteNamespace, err
+	}
+
+	if serviceDisc != nil {
+		// Try to authorize against the ServiceImport resource as we know the CRD should exist and the credentials
+		// should allow read access.
+		restConfig, _, err := resource.GetAuthorizedRestConfig(serviceDisc.Spec.BrokerK8sApiServer, serviceDisc.Spec.BrokerK8sApiServerToken,
+			serviceDisc.Spec.BrokerK8sCA, rest.TLSClientConfig{}, schema.GroupVersionResource{
+				Group:    "multicluster.x-k8s.io",
+				Version:  "v1alpha1",
+				Resource: "serviceimports",
+			}, serviceDisc.Spec.BrokerK8sRemoteNamespace)
+
+		return restConfig, serviceDisc.Spec.BrokerK8sRemoteNamespace, err
+	}
+
+	return nil, "", nil
+}
+
+func GetClientConfigAndClusterName(rules *clientcmd.ClientConfigLoadingRules, overrides *clientcmd.ConfigOverrides) (RestConfig, error) {
```

Comment:
```suggestion
func clientConfigAndClusterName(rules *clientcmd.ClientConfigLoadingRules, overrides *clientcmd.ConfigOverrides) (RestConfig, error) {
```

No need to export it, it's only used internally